### PR TITLE
Create log-in state

### PR DESCRIPTION
**Pull request by yairtawil**

### CODE REVIEW

File: src/app/packages/login/helpers/ansyn-jwt.class.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import HmacSHA256 from 'crypto-js/hmac-sha256';
```
Comment (MarkShleifer) [inline src/app/packages/login/helpers/ansyn-jwt.class.ts]: RMV

File: src/app/packages/login/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	constructor(private authService: AuthService,
+				private activatedRoute: ActivatedRoute,
+				private router: Router) { }
+
+	ngOnInit(): void {
+		this.returnUrl$.subscribe(_returnUrl => {
+			this.returnUrl = _returnUrl;
+		});
+		this.authService.clear()
```
Comment (MarkShleifer) [inline src/app/packages/login/login/login.component.ts]: ;

File: src/app/packages/login/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	get login$() {
```
Comment (MarkShleifer) [inline src/app/packages/login/login/login.component.ts]: rethink it

File: src/app/packages/login/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+})
+
+export class LoginComponent implements OnInit {
+
+	username: string;
+	password: string;
+	rememberMe: boolean;
+	returnUrl = this.authService.authrizedPath;
+	returnUrl$: Observable<string> = this.activatedRoute
+		.queryParams
```
Comment (MarkShleifer) [inline src/app/packages/login/login/login.component.ts]: https://angular.io/guide/router#activatedroute-the-one-stop-shop-for-route-information

File: src/app/ansyn/ansyn/ansyn.component.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
+import { IAppState } from '../../app-reducers';
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/ansyn/ansyn/ansyn.component.ts]: Application

**Pull request merged**